fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java

```diff
@@ -64,38 +64,55 @@ public HiveRemoteFileIO(Configuration configuration) {
         ImmutableMap.Builder<RemotePathKey, List<RemoteFileDesc>> resultPartitions = ImmutableMap.builder();
         String path = ObjectStorageUtils.formatObjectStoragePath(pathKey.getPath());
         List<RemoteFileDesc> fileDescs = Lists.newArrayList();
         try {
             URI uri = new Path(path).toUri();
             FileSystem fileSystem;
-
             if (!FeConstants.runningUnitTest) {
                 fileSystem = FileSystem.get(uri, configuration);
             } else {
                 fileSystem = this.fileSystem;
             }
-
-            RemoteIterator<LocatedFileStatus> blockIterator;
-            if (!pathKey.isRecursive()) {
-                blockIterator = fileSystem.listLocatedStatus(new Path(uri.getPath()));
+            List<Path> expandedPaths = Lists.newArrayList();
+            if (!expandWildCards) {
+                expandedPaths.add(new Path(uri.getPath()));
             } else {
-                blockIterator = listFilesRecursive(fileSystem, new Path(uri.getPath()));
+                FileStatus[] status = fileSystem.globStatus(new Path(uri.getPath()));
+                for (FileStatus s : status) {
+                    expandedPaths.add(s.getPath());
+                }
             }
-            while (blockIterator.hasNext()) {
-                LocatedFileStatus locatedFileStatus = blockIterator.next();
-                if (!isValidDataFile(locatedFileStatus)) {
-                    continue;
+            for (Path expandedPath : expandedPaths) {
+                RemoteIterator<LocatedFileStatus> blockIterator;
+                if (!pathKey.isRecursive()) {
+                    blockIterator = fileSystem.listLocatedStatus(expandedPath);
+                } else {
+                    blockIterator = listFilesRecursive(fileSystem, expandedPath);
+                }
+                while (blockIterator.hasNext()) {
+                    LocatedFileStatus locatedFileStatus = blockIterator.next();
+                    if (!isValidDataFile(locatedFileStatus)) {
+                        continue;
+                    }
+                    String locateName = locatedFileStatus.getPath().toUri().getPath();
+                    String fileName = PartitionUtil.getSuffixName(expandedPath.toUri().getPath(), locateName);
+
+                    BlockLocation[] blockLocations = locatedFileStatus.getBlockLocations();
+                    List<RemoteFileBlockDesc> fileBlockDescs = getRemoteFileBlockDesc(blockLocations);
+                    RemoteFileDesc fileDesc = new RemoteFileDesc(fileName, "", locatedFileStatus.getLen(),
+                            locatedFileStatus.getModificationTime(), ImmutableList.copyOf(fileBlockDescs),
+                            ImmutableList.of());
+                    if (expandWildCards) {
+                        fileDesc.setFullPath(locatedFileStatus.getPath().toString());
```

Comment:
why when expandWildCards=true needs to setFullPath

Reply:
See the change in `fe/fe-core/src/main/java/com/starrocks/planner/FileTableScanNode.java`

Basically when `wildcards` is in tableLocation, we cannot simply concatenate the `tableLocation` + `fileName` to get the FullPath as `tableLocation` might contain wildcards. Thus we just pass down the fullPaths directly.